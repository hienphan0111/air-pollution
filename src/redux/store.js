import { configureStore } from '@reduxjs/toolkit';
import countriesReducer from './countries';
import citiesReducer from './cities';
// import cityReducer from '@/routes/city-detail';
import cityReducer from './city';

const store = configureStore({
  reducer: {
    countries: countriesReducer,
    cities: citiesReducer,
    cityInfo: cityReducer,
  },
});

export default store;
