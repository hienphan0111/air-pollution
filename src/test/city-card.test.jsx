/**
 * @jest-environment jsdom
 */

import React from 'react';
import TestRenderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect.js';
import { Provider, useDispatch, useSelector } from 'react-redux';
import store from '../redux/store';
import CityCard from '../components/city-card.component';
import axios from 'axios';
import { render, screen, fireEvent } from '@testing-library/react';
import { BrowserRouter, MemoryRouter } from 'react-router-dom';

jest.mock('axios', () => ({
  post: jest.fn(),
  get: jest.fn(),
}));

const reactRedux = { useDispatch, useSelector };

describe('Render the detail info of city', () => {
  
  const useSelectorMock = jest.spyOn(reactRedux, 'useSelector')
  const useDispatchMock = jest.spyOn(reactRedux, 'useDispatch')

  beforeEach(() => {
    useDispatchMock.mockClear();
    useSelectorMock.mockClear();
  })
  afterEach(() => {
      useDispatchMock.mockClear();
      useSelectorMock.mockClear();
  })

  const city = {
    id: 5,
    name: 'one',
    aqi: 10,
    time: {
      s: '2022-21-12 17:00'
    },
  }
  
  it('Render detail info of city', () => {

    // useSelectorMock.mockImplementation(selector => selector(mockStore));
    useDispatchMock.mockImplementation(() => () => {});
    // const data = useSelectorMock.mockImplementation(() => mockStore);

    const card = TestRenderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <CityCard key={city.id} city={city}/>
        </MemoryRouter>
      </Provider>
    ).toJSON();
  
    expect(card).toMatchSnapshot();

  })

});