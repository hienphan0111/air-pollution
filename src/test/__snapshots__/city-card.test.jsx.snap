// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display city info on Card Render detail info of city 1`] = `
<div
  className="city-card"
>
  <div
    className="city-info"
  >
    <h3>
      one
    </h3>
    <span>
      2022-21-12 17:00
    </span>
  </div>
  <div
    className="aqi"
  >
    <div
      className="aqi-info"
    >
      <h3
        style={
          {
            "backgroundColor": "#0c563b",
          }
        }
      >
        10
      </h3>
      <span
        className="title"
      >
        Good
      </span>
    </div>
    <a
      href="/city-detail"
      onClick={[Function]}
    >
      <div
        className="next-city"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <svg
          className="w-6 h-6 next-icon"
          fill="none"
          stroke="currentColor"
          strokeWidth={1.5}
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12.75 15l3-3m0 0l-3-3m3 3h-7.5M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
            strokeLinecap="round"
            strokeLinejoin="round"
          />
        </svg>
      </div>
    </a>
  </div>
</div>
`;
